codegen/src/test/resources/software/amazon/awssdk/codegen/poet/rules/endpoint-resolve-interceptor-preSra.java

```diff
@@ -98,12 +100,27 @@ public static QueryEndpointParams ruleParams(SdkRequest request, ExecutionAttrib
         builder.region(AwsEndpointProviderUtils.regionBuiltIn(executionAttributes));
         builder.useDualStackEndpoint(AwsEndpointProviderUtils.dualStackEnabledBuiltIn(executionAttributes));
         builder.useFipsEndpoint(AwsEndpointProviderUtils.fipsEnabledBuiltIn(executionAttributes));
+        builder.credentialScope(getCredentialScopeIfPresent(executionAttributes));
         setClientContextParams(builder, executionAttributes);
         setContextParams(builder, executionAttributes.getAttribute(AwsExecutionAttribute.OPERATION_NAME), request);
         setStaticContextParams(builder, executionAttributes.getAttribute(AwsExecutionAttribute.OPERATION_NAME));
         return builder.build();
     }
 
+    private static Region getCredentialScopeIfPresent(ExecutionAttributes executionAttributes) {
+        SelectedAuthScheme<?> authScheme = executionAttributes.getAttribute(SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME);
+        try {
+            Identity identity = authScheme.identity().get();
```

Comment:
Can we avoid this blocking call?

Reply:
discussed offline, will use joinLikeSync()